pkg/utils/configMapUtils.go

```diff
@@ -10,6 +11,57 @@ import (
 	"k8s.io/client-go/rest"
 )
 
+// ValidateConfigMaps validates the configMap, before checking or creating them.
+func ValidateConfigMaps(configMaps []v1alpha1.ConfigMap, engineDetails EngineDetails, clients ClientSets) ([]v1alpha1.ConfigMap, []error) {
+
+	var errorList []error
+	var validConfigMaps []v1alpha1.ConfigMap
+
+	for _, v := range configMaps {
+		if v.Name == "" || v.MountPath == "" {
+			log.Infof("Unable to validate the configMap, with Name: %v , with mountPath: %v", v.Name, v.MountPath)
+			e := errors.New("Aborting Execution, configMap Name or mountPath is invalid")
+			errorList = append(errorList, e)
+			return nil, errorList
+		}
+
+		_, err := clients.KubeClient.CoreV1().ConfigMaps(engineDetails.AppNamespace).Get(v.Name, metav1.GetOptions{})
+
+		if err != nil {
+			//errors = append(errors, err)
+			log.Infof("Unable to find ConfigMap with Name: %v", v.Name)
+
+			if v.Data != nil {
+				log.Infof("Will try to build configMap with Name : %v", v.Name)
+				configMapObject := createConfigMapObject(v)
+
+				_, err = clients.KubeClient.CoreV1().ConfigMaps(engineDetails.AppNamespace).Create(configMapObject)
+
+				if err != nil {
+					log.Errorf("Unable to create ConfigMap Error : %v", err)
+					errorList = append(errorList, err)
+					return nil, errorList
+				}
+				validConfigMaps = append(validConfigMaps, v)
+				log.Infof("Successfully created ConfigMap with Name: %v", v.Name)
+
+			} else {
+				log.Infof("Did'nt find the configMap with Name: %v, and Data is also empty. Aborting Execution", v.Name)
```

Comment:
a bit confused here. Are we saying configmap found in spec but there is no such resource in cluster and you are unable to create as data is empty?

Reply:
If so, can this log be improved a bit..

```suggestion
				log.Infof("configMap with name: %v not found. unable to create this configMap as no data is specified. Aborting Execution", v.Name)
```